### PR TITLE
Publish Release workflow

### PR DESCRIPTION
**Pull request by Jaybsoni**

### CODE REVIEW

File: .github/workflows/upload.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Run tests
+        run: |
+          python -m pytest tests --tb=native
+
+      - name: Publish
+        uses: pypa/gh-action-pypi-publish@master
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI }}
```
Comment (Jaybsoni) [inline .github/workflows/upload.yml:36]: Probably need to setup this token for the repo

File: .github/workflows/upload.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          python setup.py bdist_wheel
+          pip install dist/PennyLane*.whl
+
+      - name: Install test dependencies
+        run: |
+          pip install wheel qsimcirq pytest pytest-cov pytest-mock --upgrade
+
+      - name: Run tests
+        run: |
+          python -m pytest tests --tb=native
```
Comment (Jaybsoni) [inline .github/workflows/upload.yml:30]: No tests to run ...? 
```suggestion
```

File: .github/workflows/upload.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Build and install Plugin
+        run: |
+          python -m pip install --upgrade pip wheel
+          python setup.py bdist_wheel
+          pip install dist/PennyLane*.whl
+
+      - name: Install test dependencies
+        run: |
+          pip install wheel qsimcirq pytest pytest-cov pytest-mock --upgrade
```
Comment (Jaybsoni) [inline .github/workflows/upload.yml:26]: No test dependancies because no tests 
```suggestion
```

File: .github/workflows/upload.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          python-version: 3.7
+
+      - name: Build and install Plugin
+        run: |
+          python -m pip install --upgrade pip wheel
+          python setup.py bdist_wheel
+          pip install dist/PennyLane*.whl
+
+
```
Comment (antalszava) [inline .github/workflows/upload.yml:24]:
```suggestion
```

Comment (Jaybsoni) [inline .github/workflows/upload.yml:24]:
```suggestion
```

File: .github/workflows/upload.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+
+      - name: Build and install Plugin
+        run: |
+          python -m pip install --upgrade pip wheel
+          python setup.py bdist_wheel
+          pip install dist/PennyLane*.whl
```
Comment (antalszava) [inline .github/workflows/upload.yml:22]: Likely we could even skip this step, as we're not running the test suite (hence the plugin doesn't have to be installed).
```suggestion
```

File: .github/workflows/upload.yml (hunk 1 above)
Comment (antalszava) [reply .github/workflows/upload.yml:28]: :+1: Will check with Josh on this.

**Review by antalszava — APPROVED** [review]
Looks good to me! :+1: Thank you :slightly_smiling_face:

File: .github/workflows/upload.yml (hunk 1 above)
Comment (josh146) [reply .github/workflows/upload.yml:27]: done :)

Comment (antalszava) [reply .github/workflows/upload.yml:27]: Thank you!

**Pull request merged**